pandas/_libs/groupby.pyx

```diff
@@ -779,9 +779,10 @@ def group_quantile(ndarray[float64_t] out,
                 non_na_counts[lab] += 1
 
     # Get an index of values sorted by labels and then values
-    order = (values, labels)
-    sort_arr = np.lexsort(order).astype(np.int64, copy=False)
-
+    sort_arr = np.arange(len(labels), dtype=np.int64)
```

Comment:
Is there a reason for this to be arange? Does that not impact the ordering at all?

Reply:
@dsaxton

Reply:
I tried a new approach: remove the `NaN` labels (`-1`) from the lower part before `np.lexsort`, so when we make the calculations it's on the upper side. Since we iterate through the number of groups `ngroups` and the `NaN` group is not counted, so the `NaN` labels will not affect the final result.